src/Tribe/Views/V2/Views/Month_View.php

```diff
@@ -455,4 +365,59 @@ protected function calculate_grid_start_end( $date ) {
 		return [ Dates::build_date_object( $grid_start ), Dates::build_date_object( $grid_end ) ];
 	}
 
+	/**
+	 * Builds the next or prev URL given the date that should be used.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed $date          The date to build the URL from, a date object or string.
+	 * @param bool  $canonical     Whether to return the canonical version of the URL or not.
+	 * @param array $passthru_vars An array of variables that should be preserved and applied to the resulting URL.
+	 *
+	 * @return string The URL as built from the event.
+	 */
+	protected function build_url_for_date( $date, $canonical, array $passthru_vars = [] ) {
+		$url = $this->get_url();
+		$date = Dates::build_date_object( $date );
+
+		$event_date_aliases = $this->url->get_query_args_aliases_of( 'event_date', $this->context );
+		$event_date_aliases = array_unique( array_merge( $event_date_aliases, [ 'eventDate', 'tribe-bar-date' ] ) );
+
+		if ( ! empty( $event_date_aliases ) ) {
+			$url = remove_query_arg( $event_date_aliases, $this->get_url() );
+		}
+
+		$url = add_query_arg( [ 'eventDate' => $date->format( 'Y-m' ) ], $url );
+
+		if ( ! empty( $url ) && $canonical ) {
+			$input_url = $url;
+
+			if ( ! empty( $passthru_vars ) ) {
+				$input_url = remove_query_arg( array_keys( $passthru_vars ), $url );
+			}
+
+			// Make sure the view slug is always set to correctly match rewrites.
+			$input_url = add_query_arg( [ 'eventDisplay' => $this->slug ], $input_url );
+
+			$canonical_url = tribe( 'events.rewrite' )->get_clean_url( $input_url );
+
+			if ( ! empty( $passthru_vars ) ) {
+				$canonical_url = add_query_arg( $passthru_vars, $canonical_url );
+			}
+
+			$url = $canonical_url;
+		}
+
+		return $url;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @since TBD
+	 */
+	protected function get_label_format() {
+		// Something link "January".
+		return 'F';
```

Comment:
Here as well. Maybe add a filter or at least a `@todo` to discuss it later.

Reply:
I've resolved by filtering the label in the `View::get_link_label` method.